ocaml/typing/ctype.ml

```diff
@@ -4428,7 +4391,7 @@ let rec moregen inst_nongen variance type_pairs env t1 t2 =
         moregen_occur env (get_level t1) t2;
         update_scope_for Moregen (get_scope t1) t2;
         occur_for Moregen env t1 t2;
-        constrain_type_jkind_exn env Moregen t2 jkind;
+        check_type_jkind_exn env Moregen t2 jkind;
```

Comment:
```suggestion
        (* use [check], not [constrain], here because [constrain] would be like
        instantiating [t2], which we do not wish to do *)
        check_type_jkind_exn env Moregen t2 jkind;
```